rfd/0109-cloud-agent-upgrades.md

```diff
@@ -327,6 +327,33 @@ to be changed to follow the same pattern of registering a release channel and in
 by ec2 discovery.
 
 
+### ECS Fargate Agents
+When Teleport Discover flow is used to connect an RDS, a Teleport agent is automatically provisioned using ECS Fargate. The automatic update
+functionality for regular agents depends on systemd or Kubernetes. Fargate agents run in containers, so that means the existing automatic
+update functionality is not compatible with these agents. We need a separate method of keeping these Fargate agents updated.
+
+To keep Farage agents updated, Teleport will run a separate task that checks every 30 minutes for an available update and uses the ECS
+Fargate API to update Teleport agents. This task does not need to be run on all Teleport clusters. It will only be needed if the
```

Comment:
Which API are we going to use?

We might need to call the following APIs:
`ecs:DescribeClusters`: the cluster name is deterministic and has some tags that help us ensure it was created by Teleport using the DeployService.
`ecs:DescribeServices`: the service name is deterministic and has some tags that help us ensure it was created by Teleport using the DeployService.

We then need to extract information from the Service, in order for us to recreate it
However, we first need to register a new task definition:
`ecs:DescribeTaskDefinition`: we should get the latest definition used by the service
`ecs:RegisterTaskDefinition`: with the info above, we can create a new task definition that has the new container image version (everything else must stay the same).

Then, we should be ready to re-deploy the service.
We can call `ecs:UpdateService` with the new Task Definition, but I'm not sure it tries graceful shutdown of the container.
If not, then we need `ecs:DeleteService`, `ecs:WaitUntilServicesInactive` and then `ecs:CreateService`

Reply:
Thanks for giving some more details on the ECS APIs. I'll add them to the RFD.